projects/elfutils/build.sh

```diff
@@ -82,12 +82,39 @@ fi
 
 ASAN_OPTIONS=detect_leaks=0 make -j$(nproc) V=1
 
+# External dependencies used by the fuzz targets have to be built
+# with MSan explictily to avoid bogus "security" bug reports like
+# https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=45630
+# and https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=45631.
+zlib="-l:libz.a"
+if [[ "$SANITIZER" == memory ]]; then
+    (
+    git clone https://github.com/madler/zlib
```

Comment:
Should we clone it in the Dockerfile though? Is the reason to do it here that compilation of the library is only relevant in MSan contexts?

Reply:
sounds good, will keep in mind for the future as well.

Reply:
I think I added a comment to `build.sh` describing how to do that and I have a few patches to make it a bit easier to run `build.sh` locally but nobody uses it apart from me because generally `clang` (and `libfuzzer`) aren't used by the elfutils project (and as far as I can remember clang-related patches I sent to the mailing list were rejected) so to reproduce issues `gcc`/`afl-gcc` or `gcc/valgrind` are used. Because of that issues reported by MSan usually have to be reproduced with `valgrind` manually and if it confirms those issues they are considered bugs.

Reply:
I'm not sure if it's helpful or not but just in case those clang-related patches were discussed in https://sourceware.org/pipermail/elfutils-devel/2021q4/004456.html and https://sourceware.org/pipermail/elfutils-devel/2021q4/004468.html. On the bright side ASan was integrated into the elfutils CI along the way.

Reply:
okay -- thanks for the summary and the links here (https://sourceware.org/pipermail/elfutils-devel/2021q4/004456.html) is definitely u